crane/container.go

```diff
@@ -122,7 +122,9 @@ func (c *container) Dependencies() *Dependencies {
 	dependencies := &Dependencies{}
 	for _, link := range c.RunParams.Link() {
 		linkParts = strings.Split(link, ":")
-		dependencies.All = append(dependencies.All, linkParts[0])
+		if !dependencies.includesAsKind(linkParts[0], "link") {
```

Comment:
Agree with all your points, but I think it's not relevant to our discussion - I think you misinterpreted the semantics of `Dependencies` (probably due to a lack of documentation!).

The actual usage of multiple aliases does not rely on `Dependencies`, but on `Links()` https://github.com/michaelsauter/crane/blob/0e58aa4202647770d2f4484876331712369a8f8a/crane/container.go#L533-L536.

If there was a built-in `Set` in golang, members of `Dependencies` would be of that type - there is no benefit in having duplicated entries (as you have noticed by hitting the bug in the first place) - check throughout the code, it's only used to append and check for existence/"contains". So as you were about to prevent duplicate in the `.All` member, I thought you could also do it in `.Link`, as it's done for the other source of dependencies such as `--volumes-from` or `--net container:<foo>`